### PR TITLE
Make Java 1.7 the minimum supported version

### PR DESCRIPTION
**Pull request by rleigh-dundee**
- Update minimum version of JVM and bytecode to 1.7.
- Replace travis testing of openjdk6 with openjdk7 (this could be dropped to save time--we already test with oraclejdk7)

Note that the main impact of this PR will be to ImageJ users who are using a 1.6 JVM as bundled with ImageJ or Fiji.  However, both projects are working on updating this so should be resolved by the time 5.2 is released.

--------

Testing:

This should be taken care of by the CI builds and full repo testing.

### CODE REVIEW

**Comment by rleigh-dundee** [comment]
--no-rebase

**Comment by rleigh-dundee** [comment]
@melissalinkert I opened this PR to match the openmicroscopy equivalent, however I didn't realise we hadn't also created a dev_5_1 branch at this point in time since I intended this to be a 5.2 change.  I can close it and reopen at a future time if that's more appropriate.  On the other hand, if we are going to branch dev_5_1 sometime soon, I could leave it open for merging after that point.

**Comment by melissalinkert** [comment]
Might be nice to take this opportunity to use a ```java.version``` property or similar, so that future upgrades don't require changing every ```build.properties``` file.

**Comment by joshmoore** [comment]
Moving to breaking since targeting the next major version.

**Comment by sbesson** [comment]
@rleigh-dundee can you merge `origin/develop` into this branch to fix the conflict?

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [BIOFORMATS-DEV-merge-push#6](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-push/6/). See the [console output](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-push/6/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - .travis.yml

**Comment by rleigh-dundee** [comment]
@sbesson Done

File: pom.xml [hunk 1]
```diff
@@ -319,8 +319,8 @@
           <version>3.1</version>
           <!-- Require the Java 6 platform. -->
```
Comment (mtbc) [inline pom.xml]: Java 7

**Comment by mtbc** [comment]
These commits look generally good. They leave a fair few `http://docs.oracle.com/javase/6/docs/`-based URLs around; I'd guess we should also update many of those to 7?

**Comment by rleigh-dundee** [comment]
@mtbc These should now be corrected.

**Comment by mtbc** [comment]
Thank you. :+1: Good to merge.

**Comment by sbesson** [comment]
Discussed at standup, the only follow up might be to remove the `openjdk7` or `oraclejdk7` component from the Travis matrix assuming this is tested by one the CI combinations. Otherwise, all looks good. Merging.

**Pull request merged**